### PR TITLE
Fix graphql plugin resolverof

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description of what you did:
Fix https://github.com/strapi/strapi/issues/4035

#### My PR is a:

- [ ] 💥 Breaking change
- [X] 🐛 Bug fix
- [ ] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [X] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/controllers/User.js [hunk 1]
```diff
@@ -170,27 +170,36 @@ module.exports = {
...
     const { email, username, password } = ctx.request.body;
 
-    if (!email) return ctx.badRequest('missing.email');
-    if (!username) return ctx.badRequest('missing.username');
-    if (!password) return ctx.badRequest('missing.password');
-
-    const userWithSameUsername = await strapi
-      .query('user', 'users-permissions')
-      .findOne({ username });
+    // if (!email) return ctx.badRequest('missing.email');
```
Comment (lauriejim) [inline packages/strapi-plugin-users-permissions/controllers/User.js]: Can you remove these unsued lines

Comment (alexandrebodin) [reply packages/strapi-plugin-users-permissions/controllers/User.js]: sure !

**Comment by avirankatzsofi** [comment]
After updating to beta.16.15, I see that any user can update any other user (given that the updating user is permitted to use `updateUser`). Is that intended?

**Comment by alexandrebodin** [comment]
@avirankatzsofi do you mean user or admin ?
if you make the updateUser permission public or authenticated then sadly it will be acessible by the users having the role. There is no notion of context for those routes. if you want a user to be able to update itself you should make a custom mutation for then and those ones are mainly designed to be for  administration

**Pull request merged**